sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -134,6 +134,7 @@ object Cast {
     case (DateType, TimestampType) => true
     case (_: AtomicType, StringType) => true
     case (_: CalendarIntervalType, StringType) => true
+    case (NullType, StringType) => true
```

Comment:
So this is a logical partial recover of the previous code `case (_, StringType) => true` before #25242, right?

Reply:
I think we can cast null type to any type, how about `case (NullType, _) => true`?

Reply:
+1, @cloud-fan 's comment.

In that case, let's create a new JIRA instead of FOLLOWUP, @jiangxb1987 .

Reply:
+1, @cloud-fan 's comment.